cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	if req.Pagination != nil && req.Pagination.Limit > maxLimit {
+		return fmt.Errorf("limit must not exceed %d", maxLimit)
+	}
+
+	return nil
+}
+
+type transactionInfo struct {
+	// Successful indicates whether the transaction was successful or not
+	Successful bool `json:"status"`
+	// ApplicationOrder is the index of the transaction among all the transactions
+	// for that ledger.
+	ApplicationOrder int32 `json:"applicationOrder,omitempty"`
+	// FeeBump indicates whether the transaction is a feebump transaction
+	FeeBump bool `json:"feeBump,omitempty"`
+	// EnvelopeXdr is the TransactionEnvelope XDR value.
+	EnvelopeXdr string `json:"envelopeXdr,omitempty"`
+	// ResultXdr is the TransactionResult XDR value.
+	ResultXdr string `json:"resultXdr,omitempty"`
+	// ResultMetaXdr is the TransactionMeta XDR value.
+	ResultMetaXdr string `json:"resultMetaXdr,omitempty"`
+	// DiagnosticEventsXDR is present only if transaction was not successful.
+	// DiagnosticEventsXDR is a base64-encoded slice of xdr.DiagnosticEvent
+	DiagnosticEventsXDR []string `json:"diagnosticEventsXdr,omitempty"`
+	// Ledger is the sequence of the ledger which included the transaction.
+	Ledger uint32 `json:"ledger,omitempty"`
+	// LedgerCloseTime is the unix timestamp of when the transaction was included in the ledger.
+	LedgerCloseTime int64 `json:"createdAt,string,omitempty"`
+}
+
+// GetTransactionsResponse encapsulates the response structure for getTransactions queries.
+type GetTransactionsResponse struct {
+	Transactions          []transactionInfo `json:"transactions"`
+	LatestLedger          uint32            `json:"latestLedger"`
+	LatestLedgerCloseTime int64             `json:"latestLedgerCloseTimestamp"`
+	OldestLedger          uint32            `json:"oldestLedger"`
+	OldestLedgerCloseTime int64             `json:"oldestLedgerCloseTimestamp"`
+	Cursor                string            `json:"cursor"`
+}
+
+type transactionsRPCHandler struct {
+	ledgerReader      db.LedgerReader
+	dbReader          db.TransactionReader
+	maxLimit          uint
+	defaultLimit      uint
+	logger            *log.Entry
+	networkPassphrase string
+}
+
+// getTransactionsByLedgerSequence fetches transactions between the start and end ledgers, inclusive of both.
+// The number of ledgers returned can be tuned using the pagination options - cursor and limit.
+func (h transactionsRPCHandler) getTransactionsByLedgerSequence(ctx context.Context, request GetTransactionsRequest) (GetTransactionsResponse, error) {
+	ledgerRange, err := h.dbReader.GetLedgerRange(ctx)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	err = request.isValid(h.maxLimit, ledgerRange)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	// Move start to pagination cursor
+	start := toid.New(int32(request.StartLedger), 1, 0)
```

Comment:
operation index should likewise start at 1, this is defined in the [TOID spec](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0035.md): 

> **Operation Index:** the index of the operation within its parent transaction. This takes up 12 bits of the id, from bits 52 to 63, inclusive. The operation index value starts at 1. The maximum supported number of operations per transaction is 4095.